File: contest/ad_infinitum13/IntegralPoints.java
import java.math.BigInteger;
import java.util.Scanner;

// FIXME something wrong with order of BigIntegers or other issue. Check RubyCode which passed.
public class IntegralPoints {
    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int tests = stdin.nextInt();
        for(int i = 0; i < tests; i++) {
            int x1 = stdin.nextInt();
            int y1 = stdin.nextInt();
            int x2 = stdin.nextInt();
            int y2 = stdin.nextInt();
            int x3 = stdin.nextInt();
            int y3 = stdin.nextInt();

            long outerPoints = outer(x1, y1, x2, y2);
            outerPoints += outer(x1, y1, x3, y3);
            outerPoints += outer(x2, y2, x3, y3);
            outerPoints += 3;

            BigInteger area = BigInteger.valueOf((long) x1).subtract(BigInteger.valueOf((long) x3))
                    .multiply(BigInteger.valueOf((long) y2).subtract(BigInteger.valueOf((long) y1)))

                    .subtract(

                            BigInteger.valueOf((long) x1).subtract(BigInteger.valueOf((long) x2))
                                    .multiply(BigInteger.valueOf((long) y3).subtract(BigInteger.valueOf((long) y1))));

            BigInteger interiorPoints = ((area.subtract(BigInteger.valueOf(outerPoints))).add(BigInteger.valueOf(2L))).divide(BigInteger.valueOf(2L)).max(BigInteger.ZERO);
            System.out.println(interiorPoints);
        }
        stdin.close();
    }

    private static long outer(int x1, int y1, int x2, int y2) {
        return Math.max(gcd(Math.abs(x2 - x1), Math.abs(y2 - y1)) - 1, 0);
    }

    private static long gcd(int x, int y) {
        if(y == 0) {
            return x;
        }
        return gcd(y, x % y);
    }
}

File: contest/ad_infinitum13/integral_points.rb
def vectorLength(x1, y1, x2, y2)
  ((x2 - x1).power(2) + (y2 - y1).power(2)).sqrt(2)
end

def gcd(x, y)
  return x if(y == 0)
  return gcd(y, x % y)
end
   
def outer(x1, y1, x2, y2)
  return [gcd((x2 - x1).abs, (y2 - y1).abs) - 1, 0].max
end

tests = gets.to_i
tests.times do |i|
  arr = gets.split(' ').map {|s| s.to_i }
  x1 = arr[0]
  y1 = arr[1]
  x2 = arr[2]
  y2 = arr[3]
  x3 = arr[4]
  y3 = arr[5]
  
  outerPoints = outer(x1, y1, x2, y2);
  outerPoints += outer(x1, y1, x3, y3);
  outerPoints += outer(x2, y2, x3, y3);
  outerPoints += 3;
  
  area = ((x1 - x3) * (y2 - y1) - (x1 - x2) * (y3 - y1)).abs;
  interiorPoints = [((area - outerPoints + 2) / 2), 0].max;
  
  puts interiorPoints
end